### PR TITLE
Profile screen fix

### PR DESCRIPTION
**Pull request by npfrei**
Fix to decrease the complexity of the ProfileScreen under 50.  It mostly consist of creating sub composables fuctions to reduce the number of nested ifs/whens. I did not add any tests as it was already tested previously.

### CODE REVIEW

File: app/src/main/java/com/lastaoutdoor/lasta/ui/screen/profile/ProfileScreen.kt [hunk 1]
```diff
@@ -232,52 +232,7 @@ fun ChangeBio(
...
-                    }
-                    Spacer(modifier = Modifier.width(16.dp))
-                    Button(onClick = { onBioChange(text) }) {
-                      Text("Save" /* TODO : make this a xml string */)
-                    }
-                  }
-                }
-          }
-        }
+    BottomSheetProfile(onDismissRequest, bioText, onBioChange)
```
Comment (JeremyHugentobler) [inline app/src/main/java/com/lastaoutdoor/lasta/ui/screen/profile/ProfileScreen.kt:235]: The previous version was way too long and it was the right call to extract it in a composable

File: app/src/main/java/com/lastaoutdoor/lasta/ui/screen/profile/ProfileScreen.kt [hunk 2]
```diff
@@ -531,3 +405,153 @@ fun RecentActivities(
...
+          Text("Bio" /* TODO : make this a xml string */)
+          Spacer(modifier = Modifier.height(8.dp))
+          TextField(
+              value = text,
+              onValueChange = {
+                if (it.length <= maxCharCount) {
+                  text = it
+                }
+              },
+              placeholder = { Text("Enter your bio...") /* TODO : make this a xml string */ },
```
Comment (JeremyHugentobler) [inline app/src/main/java/com/lastaoutdoor/lasta/ui/screen/profile/ProfileScreen.kt:494]: This should be translated

File: app/src/main/java/com/lastaoutdoor/lasta/ui/screen/profile/ProfileScreen.kt [hunk 3]
```diff
@@ -531,3 +405,153 @@ fun RecentActivities(
...
+              trailingIcon = { TrailingIcon(text = text, max = maxCharCount) },
+              modifier = Modifier.fillMaxWidth().testTag("ProfileTextField"))
+
+          Spacer(modifier = Modifier.height(16.dp))
+          Box(
+              contentAlignment = Alignment.Center,
+              modifier = Modifier.fillMaxWidth().testTag("ProfileBox")) {
+                Row {
+                  Button(onClick = onDismissRequest) {
+                    Text("Cancel" /* TODO : make this a xml string */)
```
Comment (JeremyHugentobler) [inline app/src/main/java/com/lastaoutdoor/lasta/ui/screen/profile/ProfileScreen.kt:507]: This should also be translated

File: app/src/main/java/com/lastaoutdoor/lasta/ui/screen/profile/ProfileScreen.kt [hunk 4]
```diff
@@ -531,3 +405,153 @@ fun RecentActivities(
...
+                  }
+                  Spacer(modifier = Modifier.width(16.dp))
+                  Button(onClick = { onBioChange(text) }) {
+                    Text("Save" /* TODO : make this a xml string */)
```
Comment (JeremyHugentobler) [inline app/src/main/java/com/lastaoutdoor/lasta/ui/screen/profile/ProfileScreen.kt:511]: +1

**Review by JeremyHugentobler — CHANGES_REQUESTED** [review]
Hi,

Thank you, this improves readability and reduces the complexity of our code. You did a good job extracting the different parts of the previous code into composable but there are some strings that are not translated.

I left some comments where the strings are,  but otherwise it's a nice PR and LGTM

Best,
Jérémy

File: app/src/main/java/com/lastaoutdoor/lasta/ui/screen/profile/ProfileScreen.kt (hunk 2 above)
Comment (npfrei) [reply app/src/main/java/com/lastaoutdoor/lasta/ui/screen/profile/ProfileScreen.kt]: Thanks for pointing that out !

**Review by JeremyHugentobler — APPROVED** [review]
Thank you for the quick changes ! 
See you tomorrow

**Pull request merged**